src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/TypeGVMEntriesNode.cs

```diff
@@ -0,0 +1,116 @@
+        {
+            Debug.Assert(!associatedType.IsRuntimeDeterminedSubtype);
+            Debug.Assert(TypeNeedsGVMTableEntries(associatedType));
+            _associatedType = associatedType;
+        }
+
+        public override bool HasConditionalStaticDependencies => false;
+        public override bool HasDynamicDependencies => false;
+        public override bool InterestingForDynamicDependencyAnalysis => false;
+        public override bool StaticDependenciesAreComputed => true;
+        protected override string GetName() => null;
+        public override IEnumerable<CombinedDependencyListEntry> GetConditionalStaticDependencies(NodeFactory context)
+        {
+            return Array.Empty<CombinedDependencyListEntry>();
+        }
+        public override IEnumerable<CombinedDependencyListEntry> SearchDynamicDependencies(List<DependencyNodeCore<NodeFactory>> markedNodes, int firstNode, NodeFactory context)
+        {
+            return Array.Empty<CombinedDependencyListEntry>();
+        }
+
+        public override IEnumerable<DependencyListEntry> GetStaticDependencies(NodeFactory context)
+        {
+            if (_staticDependencies == null)
+            {
+                _staticDependencies = new DependencyList();
+
+                ScanForGenericVirtualMethodEntries(context, TrackGVMDependencies);
+                ScanForInterfaceGenericVirtualMethodEntries(context, TrackInterfaceGVMDependencies);
+            }
+
+            return _staticDependencies;
+        }
+
+        public static bool TypeNeedsGVMTableEntries(TypeDesc type)
+        {
+            if (type.Category != TypeFlags.Class && !type.IsValueType)
+                return false;
+
+            // GVM table entries are built from open type definitions, not from instantiated types.
+            if (type.HasInstantiation && !type.IsGenericDefinition)
+                return false;
+
+            return type.HasGenericVirtualMethod();
+        }
+
+        public void ScanForGenericVirtualMethodEntries(NodeFactory factory, Action<NodeFactory, MethodDesc, MethodDesc> gvmEntryFoundCallback)
+        {
+            foreach (var method in _associatedType.GetMethods())
+            {
+                if (!method.IsVirtual || !method.HasInstantiation)
+                    continue;
+
+                MethodDesc slotDecl = MetadataVirtualMethodAlgorithm.FindSlotDefiningMethodForVirtualMethod(method);
+                Debug.Assert(slotDecl != null);
+                gvmEntryFoundCallback(factory, slotDecl, method);
+            }
+        }
+
+        public void ScanForInterfaceGenericVirtualMethodEntries(NodeFactory factory, Action<NodeFactory, MethodDesc, TypeDesc, MethodDesc> gvmEntryFoundCallback)
+        {
+            foreach (var iface in _associatedType.RuntimeInterfaces)
+            {
+                foreach (var method in iface.GetMethods())
+                {
+                    if (!method.HasInstantiation)
```

Comment:
Good point! I keep forgetting about static methods on interfaces

---
In reply to: [96720291](https://github.com/dotnet/corert/pull/2521#discussion_r96720291) [](ancestors = 96720291)